### PR TITLE
Week 12 - Project movies, Isabella Markskog & Sandra Wallén

### PR DESCRIPTION
**Pull request by IsabellaM5**
**Link to our deployed project**
https://musing-noether-fb9258.netlify.app/

**Our reflections on how this week's project turned out**
We did all the blue level requirements. We also did some stretch goals on red and black level.

It was a fun and challenging project! React Router is a great feature, can't wait to deep dive more into React!

One problem that we encountered though was when we tried to move the useParams from the components MovieList and MovieDetails to App.js, it made the node_modules crash. We tried to fix it but ended up moving our code to a new repo instead (this repo).

Also, right now our site isn't responsive, which we would fix if we had more time.

**Things we'd like to have clarified or explained in more detail**
We would've like to hear more about useHistory and its different methods and use cases. For instance, we know that our component BackLink isn't working as we would've wanted it (it's not dynamic), which is something that we might have been able to fix if we were presented with more information.

/Isabella & Sandra 🐯🐯

### CODE REVIEW

File: code/src/pages/MovieDetails.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const MovieDetails = () => {
+  const [movieSpecs, setMovieSpecs] = useState([])
+
+  const {id} = useParams()
+
+  useEffect(() => {
+    fetch(API_URL_DETAILS(id))
+      .then(response => response.json())
+      .then(json => setMovieSpecs(json))
+      .catch(err => console.error(err))
```
Comment (annalenksjo) [inline code/src/pages/MovieDetails.js:16]: Maybe remove the console.err 😂

File: code/public/assets/Changa/OFL.txt [hunk 2]
```diff
@@ -0,0 +1,93 @@
+Copyright 2011 Eduardo Tunni (www.tipo.net.ar)
+
+This Font Software is licensed under the SIL Open Font License, Version 1.1.
+This license is copied below, and is also available with a FAQ at:
+http://scripts.sil.org/OFL
+
```
Comment (annalenksjo) [inline code/public/assets/Changa/OFL.txt:6]: 👍

File: README.md [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
+
+One problem that we encountered was when we tried to move the useParams from the components MovieList and MovieDetails to App.js, it made the node_modules crash. We tried to fix it but ended up moving our code to a new repo instead.
+
+If we had more time, we would implement a handle loading state and more pages to the site containing more information regarding the selected movie. Also, we would've liked to fix the link path in BackLink component to be more dynamic. Right now it's only redirecting to the popular list and not depending on which list the user was previously at. 
+
```
Comment (annalenksjo) [inline README.md:15]: Good explanatory ReadMe

**Review by annalenksjo — APPROVED** [review]
Hey! 
This is a really nice project.

The page looks great. It was fun that you did a bit of your own styling and take on it. Nice with a landing page. The landing page could have had a bit more content :) 
Clean code and descriptive naming!

Really like that the movie category gets 'marked' when you have chosen it. 

We like the reflection on what to do if you had more time. 
Maybe also do a catch error thing? Good that you pointed out the problem with the back-link :)

No problems in the console. 

Good job 🐯 !

File: code/src/pages/MovieDetails.js (hunk 1 above)
Comment (KarolCislowski) [reply code/src/pages/MovieDetails.js:16]: For Sure

File: code/src/pages/MovieDetails.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  }, [id])
+  
+  return (
+    <>
+      <TopBar id={id}/>
+      {movieSpecs.backdrop_path !== undefined && (
+        <section>
+          <div className="backdrop-img" style={{ backgroundImage: `linear-gradient(180deg, rgba(0,0,0,0) 70%, rgba(0,0,0,1) 100%), url(${URL_BACKDROP(movieSpecs.backdrop_path)})`}}>
```
Comment (KarolCislowski) [inline code/src/pages/MovieDetails.js:24]: Remember to break long lines into a few lines.

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Nice project but please remove all console outputs.

All you need to know about useHistory you can find here: [https://reactrouter.com/web/api/history](https://reactrouter.com/web/api/history).

**Comment by IsabellaM5** [comment]
Hi Karol, 

We have now removed all console outputs in accordance with request :) 

/Isabella & Sandra

**Review by KarolCislowski — APPROVED** [review]

**Pull request closed without merging**